docs/adrs/00012-sbom-pruning-cli.md

```diff
@@ -0,0 +1,243 @@
+```
+
+### Key Design Principles
+
+1. **Client-Server Separation**: CLI is a thin client, all logic remains in server
+2. **HTTP-Only Communication**: No direct database access, uses REST APIs
+3. **Standard Authentication**: OIDC tokens via `Authorization` header
+4. **Idempotent Operations**: Safe to retry, handles failures gracefully
+5. **User Safety**: Dry-run by default, explicit confirmation for large deletions
+
+## Consequences
+
+### Benefits
+
+1. **Immediate Value**: Solves storage cost problem without server changes
+2. **Lightweight**: Users can install CLI without full trustd server
+3. **Flexible Deployment**: Can run from any machine with API access
+4. **Safe**: Dry-run mode and explicit confirmation prevent accidents
+5. **Familiar Pattern**: Follows industry standards (kubectl, docker CLI)
+6. **Future-Proof**: Foundation for additional client-side commands
+7. **Testable**: Easy to test HTTP API integration
+
+### Trade-offs
+
+1. **Network Dependency**: Requires connectivity to Trustify server
+2. **Performance**: HTTP calls slower than direct database access
+3. **Authentication Burden**: Users must manage tokens
+4. **No Atomic Batch**: Deletes SBOMs one-by-one (mitigated in Phase 3 with batch API)
+
+### Risks and Mitigations
+
+**Risk**: Accidental mass deletion
+- **Mitigation**: Dry-run by default, require explicit `--yes` flag for operations >100 SBOMs
+
+**Risk**: Network failures during long-running operations
+- **Mitigation**: Implement retry logic, idempotent deletes, checkpoint progress
+
+**Risk**: Token expiration during operation
+- **Mitigation**: Token refresh support, clear error messages
+
+## Implementation Details
+
+### File Structure
+
+```
+trustify/                  # New binary
+├── Cargo.toml
+└── src/
+    ├── main.rs           # CLI entry point with clap
+    ├── commands/
+    │   └── prune.rs      # Prune command implementation
+    ├── client.rs         # HTTP client for Trustify API
+    └── auth.rs           # OIDC token handling
+```
+
+### Command-Line Interface
+
+```rust
+// trustify/src/commands/prune.rs
+#[derive(clap::Args)]
+pub struct PruneArgs {
+    /// Trustify API endpoint
+    #[arg(long, env = "TRUSTIFY_ENDPOINT")]
+    endpoint: String,
+
+    /// Authentication token
+    #[arg(long, env = "TRUSTIFY_TOKEN")]
+    token: Option<String>,
+
+    /// Filter query (e.g., "ingested<90 days ago")
+    #[arg(long)]
+    filter: String,
+
+    /// Dry run - preview without deleting
+    #[arg(long)]
+    dry_run: bool,
+
+    /// Skip confirmation prompt
+    #[arg(long)]
+    yes: bool,
+
+    /// Log file path
+    #[arg(long)]
+    log_file: Option<PathBuf>,
+
+    /// Batch size for pagination
+    #[arg(long, default_value = "100")]
+    batch_size: usize,
+
+    /// Max concurrent deletions
+    #[arg(long, default_value = "5")]
+    max_concurrent: usize,
+}
+```
+
+### Core Algorithm
+
+1. Authenticate with Trustify API using OIDC token
```

Comment:
Having users retrieve and provide OIDC tokens given the short life span (10 mins) they have will be frustrating to users and not practical for unassisted tasks (example Jobs).

For unassisted tasks like Jobs or CI I guess we can either provide an offline_token (retrieved from the trustify UI, but that requires a specific OIDC client with offline_access) or client_id + client_secret that will interact directly with the IdP to retrieve a valid token.
For standard cli usage like `oc` or `podman` users can use the Device Auth Flow (the token will be exposed by the UI) or even involve a browser and do Authentication Code Flow (I don't think that's worth it)